lbry/schema/claim.py

```diff
@@ -398,6 +420,18 @@ class Repost(BaseClaim):
 
     claim_type = Claim.REPOST
 
+    def update(self, **kwargs) -> dict:
+        claim_type = kwargs.pop('claim_type', None)
+        # Update common fields within BaseClaim.
+        kwargs = super().update(strict_update=False, **kwargs)
+        if claim_type:
+            # Remaining updates go into deletes/edits of ClaimReference.
+            kwargs = self.reference.update(claim_type, **kwargs)
+        return kwargs
```

Comment:
See PR description. This is where I have a choice of applying the update to BaseClaim or stuff it inside ClaimReference.